vehiclestorage/src/main/java/haveric/vehicleStorage/Updater.java

```diff
@@ -195,26 +192,16 @@ public static void query(CommandSender sender) {
                 // Add the user-agent to identify the program
                 conn.addRequestProperty("User-Agent", pluginName + "/v" + getCurrentVersion() + " (by haveric)");
 
-                // Read the response of the query
-                // The response will be in a JSON format, so only reading one line is necessary.
+                // Read the response of the query and parse the array of files from the query's response
                 final BufferedReader reader = new BufferedReader(new InputStreamReader(conn.getInputStream()));
-                String response = reader.readLine();
+                FileDetails[] fileDetails = new Gson().fromJson(reader, FileDetails[].class);
 
-                // Parse the array of files from the query's response
-                JSONArray array = (JSONArray) JSONValue.parse(response);
-
-                if (array.size() > 0) {
+                if (fileDetails.length > 0) {
                     // Get the newest file's details
-                    JSONObject latest = (JSONObject) array.get(array.size() - 1);
-
-                    // Get the version's title
-                    latestVersion = (String) latest.get(API_NAME_VALUE);
-
-                    // Get the version's link
-                    latestLink = (String) latest.get(API_LINK_VALUE);
+                    latestFile = fileDetails[0];
```

Comment:
This will always get the oldest(first) release. We need to get the last one instead.

Reply:
Indeed, I missed that